analyzers/src/SonarAnalyzer.Common/Rules/AspNet/RouteTemplateShouldNotStartWithSlashBase.cs

```diff
@@ -29,7 +29,7 @@ public abstract class RouteTemplateShouldNotStartWithSlashBase<TSyntaxKind>() :
     private const string MessageOnlyActions = "Change the paths of the actions of this controller to be relative and adapt the controller route accordingly.";
     private const string MessageActionsAndController = "Change the paths of the actions of this controller to be relative and add a controller route with the common prefix.";
 
-    private static readonly ImmutableArray<KnownType> RouteTemplateAttributes = ImmutableArray.Create(
+    private readonly ImmutableArray<KnownType> routeTemplateAttributes = ImmutableArray.Create(
```

Comment:
I guess the issue here comes from the fact that `RouteTemplateShouldNotStartWithSlashBase` is a common class for C# and VB.NET, and as such is parameterized by the type argument `TSyntaxKind`, which means that `static` fields like this one are instantiated for each concretization of the class.

This is a common problem we have with static fields in generic classes at common level.

What I would do, to avoid making this field non-static (that also forces `RouteAttributeTemplateArguments` to be non-static) is to move the list of attributes `RouteTemplateAttributes` to `AspNetMvcHelper`. That class is static, and non-generic. 

Moreover, this list doesn't seem to be specific to the rule, but something that can be used in other rules that need to check for MVC attributes on route templating. So `AspNetMvcHelper` seems a good place to host it.

What do you think??

Reply:
I moved it!
FYI for anyone to see this thread: there's an offline conversation on how to deal with this rule for this case.